pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -87,6 +71,63 @@ func NewDatastore(thisClusterID string, stopCh <-chan struct{}) (datastore.Datas
 	}, nil
 }
 
+func getSubmarinerClientset(host string, k8sSpec *datastoreSpecification) (*submarinerClientset.Clientset, error) {
+	if k8sSpec.Insecure {
+		// Shortcut
```

Comment:
`getSubmarinerRESTConfig` also checks `k8sSpec.Insecure` so I'm not sure what this shortcut is for - it seems redundant ...

Reply:
The comment could be clearer! The point of the shortcut is to avoid calling getSubmarinerRESTConfig twice, and testing the connection, when we’ve asked for an insecure connection. I could drop the shortcut entirely, and then we’d request an insecure connection, test it, and if it fails because of an unknown CA (which it won’t), request a new one. Actually now that I spell it out, given that we’ll never get the expected failure in the insecure case, I might as well drop the test.